engine.fhir/src/main/java/org/opencds/cqf/cql/engine/fhir/retrieve/R4FhirQueryGenerator.java

```diff
@@ -0,0 +1,163 @@
+package org.opencds.cqf.cql.engine.fhir.retrieve;
+
+import java.net.URLDecoder;
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+
+import org.hl7.fhir.instance.model.api.IBaseConformance;
+import org.hl7.fhir.instance.model.api.ICompositeType;
+import org.hl7.fhir.r4.model.CapabilityStatement;
+import org.hl7.fhir.r4.model.Coding;
+import org.hl7.fhir.r4.model.DataRequirement;
+import org.hl7.fhir.r4.model.DateTimeType;
+import org.hl7.fhir.r4.model.Duration;
+import org.hl7.fhir.r4.model.Period;
+import org.hl7.fhir.r4.model.Type;
+import org.opencds.cqf.cql.engine.elm.execution.SubtractEvaluator;
+import org.opencds.cqf.cql.engine.fhir.model.R4FhirModelResolver;
+import org.opencds.cqf.cql.engine.fhir.searchparam.SearchParameterMap;
+import org.opencds.cqf.cql.engine.fhir.searchparam.SearchParameterResolver;
+import org.opencds.cqf.cql.engine.runtime.Code;
+import org.opencds.cqf.cql.engine.runtime.DateTime;
+import org.opencds.cqf.cql.engine.runtime.Interval;
+import org.opencds.cqf.cql.engine.terminology.TerminologyProvider;
+
+import ca.uhn.fhir.context.FhirContext;
+
+public class R4FhirQueryGenerator extends BaseFhirQueryGenerator {
+    public R4FhirQueryGenerator(SearchParameterResolver searchParameterResolver, TerminologyProvider terminologyProvider, R4FhirModelResolver modelResolver) {
+        super(searchParameterResolver, terminologyProvider, modelResolver, FhirContext.forR4());
```

Comment:
Pass in the FHIRContext to the constructor. Don't create them inline. They are expensive.